README.md

```diff
@@ -5,46 +5,97 @@
+### Version 1.x
+
+v1.x of the library can be installed via NuGet:
+
+    Install-Package Imgix-CSharp
+
+
+## Basic Usage
+
+The library can be invoked by newing up UrlBuilder with your imgix domain. The URL builder can be reused to create URLs for any images on the domains it is provided.
 
 ```csharp
-var builder = new UrlBuilder("domain.imgix.net", useHttps: true);
+var builder = new UrlBuilder("domain.imgix.net");
+var parameters = new Dictionary<String, String>();
+parameters["w"] = "400";
+parameters["h"] = "300";
+Debug.Print(builder.BuildUrl("test.jpg", parameters));
+
+// Prints out:
+// https://domain.imgix.net/test.jpg?w=400&h=300
 ```
-    
-The constructor also accepts an array of domains, to support sharding:
+
+## Signed URLs
+
+To produce a signed URL, you must enable secure URLs on your source and then provide your signature key to the URL builder.
 
 ```csharp
-var builder = new UrlBuilder(new [] { "domain.imgix.net", "domain2.imgix.net" }, useHttps: true);
+var builder = new UrlBuilder("domain.imgix.net", signKey="aaAAbbBB11223344", signWithLibrary: false);
+var parameters = new Dictionary<String, String>();
+parameters["w"] = "500";
+parameters["h"] = "1000";
+Debug.Print(builder.BuildUrl("gaiman.jpg", parameters));
+
+// Prints out:
+// https://domain.imgix.net/gaiman.jpg?w=500&h=1000&s=fc4afbc39b6741560717142aeada876c
 ```
 
-The UrlBuilder uses a dictionary (of key/value strings) called "Parameters" to specify the values you want to pass to imgIX along the queryString.
+## Domain Sharded URLs
+
+Domain sharding enables you to spread image requests across multiple domains. This allows you to bypass the requests-per-host limits of browsers. We recommend 2-3 domain shards maximum if you are going to use domain sharding.
+
+In order to use domain sharding, you need to add multiple domains to your source. You then provide an array of these domains to a builder.
 
 ```csharp
-builder.Parameters.Add("w", "400");
-builder.Parameters.Add("h", "300");
+var builder = new UrlBuilder(new [] { "demos-1.imgix.net", "demos-2.imgix.net", "demos-3.imgix.net" });
+var parameters = new Dictionary<String, String>();
+parameters["w"] = "100";
+parameters["h"] = "100";
+Debug.Print(builder.BuildUrl("bridge.png", parameters));
+Debug.Print(builder.BuildUrl("flower.png", parameters));
+
+// Prints out:
+// http://demos-1.imgix.net/bridge.png?h=100&w=100
+// http://demos-2.imgix.net/flower.png?h=100&w=100
 ```
-    
-The UrlBuilder type also offers up a series of Properties:
 
-#### SignKey
-This is the private key for signing requests, as specified in your imgIX source.
+By default, shards are calculated using a checksum so that the image path always resolves to the same domain. This improves caching in the browser. However, you can supply a different strategy that cycles through domains instead. For example:
```

Comment:
Let's remove the mention of "cycle" and custom sharding strategies and just explain how Crc32 works here.

Reply:
I didn't quite follow. Are you suggesting we don't mention anything about cyclic domain sharding or elaborate more on the CRC sharding or both?

Reply:
Got it.